lib/publify_textfilter_answer_reveal.rb

```diff
@@ -0,0 +1,62 @@
+require 'digest'
+
+class PublifyApp
+  class Textfilter
+    class AnswerReveal < TextFilterPlugin::MacroPost
+      plugin_display_name "Answer Reveal"
+      plugin_description "Insert a answer reveal inline callout"
+
+      def self.help_text
+        %{
+You can use the following code to display an inline callout snippet. Example:
+
+<hr class="snippet" data-snippet-name="answerreveal" data-snippet-question="[ENTER QUESTION TEXT HERE]" data-snippet-answer="[ENTER ANSWER TEXT HERE]" data-snippet-callout-text="[ENTER CALLOUT TEXT HERE]" data-snippet-callout-url="[ENTER CALLOUT URL HERE]" />
+        }
+      end
+
+      def self.macrofilter(_, _, attrib, _)
+        question = attrib['data-snippet-text-question']
+        answer = attrib['data-snippet-text-answer']
+        callout_text = attrib['data-snippet-text-callout']
+        callout_url = attrib['data-snippet-url-callout']
+
+        answer = answer.gsub(/^/,"<p>").gsub(/\\n/,"</p><p>").gsub(/$/, "</p>")
+
+        callout_id = Digest::SHA256.hexdigest answer
+
+        "<div class=\"collapsible\">
+          <h3 class=\"collapsible__heading\">#{question}</h3>
+          <div class=\"collapsible__reveal\" data-dough-collapsable-target=\"answer_reveal_#{callout_id}\">
+            #{answer}
+
+            <hr class=\"snippet\" data-snippet-name=\"inlinecallout\" data-snippet-url=\"#{callout_url}\" data-snippet-text=\"#{callout_text}\" />
```

Comment:
I'm loving that this worked(!) - a nested text filter - i.e. callout within a callout